plugin/options.go

```diff
@@ -46,6 +46,13 @@ func WithTeam(team string) Option {
 	}
 }
 
+// WithConnectionTester can be specified by a plugin to enable explicit connection testing, given a spec.
+func WithConnectionTester(tester ConnectionTester) Option {
```

Comment:
Having this as an option (rather than an optional interface under `plugin.Client`, like how `OnBeforeSender` was implemented) enables us to directly call the given function without creating/instantiating the plugin with `Configure` (or `client.New`) and makes things much simpler.

Reply:
Don't we want to fallback to `Init` when the test connection is not implemented? How would we do this in this approach?

Reply:
This would be done in the CLI code once it detects the return value from this function is either an empty falsy value (in the case of an outdated SDK in the plugin) or `ErrTestConnUnimplemented`, to keep this code here simple and hassle free.

Reply:
OK then I'll take a look at the CLI code once it's ready